src/vos/evtree.c

```diff
@@ -848,12 +850,31 @@ evt_find_visible(struct evt_context *tcx, const struct evt_filter *filter,
 		return 0;
 
 	if (removals_only) {
-		/** We just want to see records not covered by a removal.  At this point,
-		 *  every record remaining in the list satisfies this condition so mark them all
-		 *  visible for sorting.
+		/** Ok, everything remaining in the list is not covered by a removal.  One more
+		 *  step is required to merge records at the same major epoch so we don't attempt
+		 *  to insert overlapping removal records.  Once this step is complete, we can sort
+		 *  the remaining records and return to the caller.
 		 */
-		d_list_for_each_entry(le, &covered, le_link) {
+		while ((le = d_list_pop_entry(&covered, struct evt_list_entry, le_link)) != NULL) {
 			this_ent = &le->le_ent;
+			d_list_for_each_entry_safe(next_le, tmp_le, &covered, le_link) {
+				next_ent = &next_le->le_ent;
+				if (next_ent->en_ext.ex_lo > this_ent->en_ext.ex_hi)
```

Comment:
It should be "next lo" > "this hi" + 1?

Reply:
this.hi + 1 == next.lo means the extents are adjacent and can be merged.  So I don't want to skip that case.

Reply:
but you break the loop and skip that case here, right?

Reply:
You are right, it is fixed now

Reply:
Ok, actually, it was not quite right.   I think it's more clear now with a bit of rearrangement but that check was checking the physical record whereas the subsequent check was checking the selected record.   Changing it as I did in prior patch you approved triggered the assertion because it wasn't checking the selected record anymore.